src/main/java/kervyn/Tasks/TaskList.java

```diff
@@ -0,0 +1,241 @@
+            if (!task.getStatus()) {
+                taskAlreadyUnMarked(dialogContainer, kervynImage);
+            } else {
+                dialogContainer.getChildren().add(
+                        DialogBox.getKervynDialog("\tOK, I've marked this task as not done yet:", kervynImage)
+                );
+                task.updateStatus();
+                dialogContainer.getChildren().add(
+                        DialogBox.getKervynDialog(task.toString(), kervynImage)
+                );
+            }
+
+            return 1;
+        }
+        catch (IndexOutOfBoundsException e) {
+            // Need to account for trying to unmark a task that doesn't exist
+            dialogContainer.getChildren().add(
+                    DialogBox.getKervynDialog("\tTask number provided doesn't exist. Please try again.", kervynImage)
+            );
+        }
+        return 0;
+    }
+
+    /**
+     * Private method to handle the scenario when a task is already marked.
+     */
+    private void taskAlreadyMarked(VBox dialogContainer, Image kervynImage) {
+        dialogContainer.getChildren().add(
+                DialogBox.getKervynDialog("\tUh oh! It looks like this task is already marked as done, please try again with another task!", kervynImage)
+        );
+    }
+
+    /**
+     * Private method to handle the scenario when a task is already unmarked.
+     */
+    private static void taskAlreadyUnMarked(VBox dialogContainer, Image kervynImage) {
+        dialogContainer.getChildren().add(
+                DialogBox.getKervynDialog("\tUh oh! It looks like this task is already marked as not done, please try again with another task!", kervynImage)
+        );
+    }
+
+    /**
+     * Removes a task from the task list.
+     *
+     * @param userTasks The ArrayList of Task objects.
+     * @param processedUserInput The user input processed into an array of Strings.
+     */
+    public void removeTask(ArrayList<Task> userTasks, String[] processedUserInput, Image kervynImage, VBox dialogContainer) {
+        try {
+            Task task = userTasks.get(Integer.parseInt(processedUserInput[1]) - 1);
+            dialogContainer.getChildren().add(
+                    DialogBox.getKervynDialog("\tOK, I've removed this task as per your request:", kervynImage)
+            );
+            dialogContainer.getChildren().add(
+                    DialogBox.getKervynDialog(task.toString(), kervynImage)
+            );
+            userTasks.remove(task);
+        }
+        catch (IndexOutOfBoundsException e) {
+            // Need to account for trying to delete a task that doesn't exist
+            dialogContainer.getChildren().add(
+                    DialogBox.getKervynDialog("\tTask number provided doesn't exist. Please try again.", kervynImage)
+            );
+        }
+    }
+
+    public void findTask(ArrayList<Task> userTasks, String userInput, Image kervynImage, VBox dialogContainer) {
```

Comment:
I think its more apparent in this class, just wanted to point out your use of similar naming conventions for related methods! e.g. `markTask`, `unMarkTask`, `findTask`, `removeTask` - naming these methods similarly does help make your code readable! good job!